cmd/zetae2e/run.go

```diff
@@ -157,3 +163,45 @@ func parseCmdArgsToE2ETestRunConfig(args []string) ([]runner.E2ETestRunConfig, e
 	}
 	return tests, nil
 }
+
+// updateConfigWithDynamicERC20 loads ERC20 addresses via gRPC given CLI flags
+func updateConfigWithDynamicERC20(ctx context.Context, conf *config.Config, erc20ChainName, erc20Symbol string) error {
+	clients, err := zetae2econfig.GetZetaClients(conf.RPCs.ZetaCoreGRPC)
+	if err != nil {
+		return fmt.Errorf("get zeta clients: %w", err)
+	}
+
+	supportedChainsRes, err := clients.ObserverClient.SupportedChains(ctx, &observertypes.QuerySupportedChains{})
+	if err != nil {
+		return fmt.Errorf("get chain params: %w", err)
+	}
+
+	chainID := int64(0)
+	for _, chain := range supportedChainsRes.Chains {
+		if chain.Name == erc20ChainName {
+			chainID = chain.ChainId
+			break
+		}
+	}
+	if chainID == 0 {
+		return fmt.Errorf("chain %s not found", erc20ChainName)
+	}
+
+	foreignCoinsRes, err := clients.FungibleClient.ForeignCoinsAll(ctx, &fungibletypes.QueryAllForeignCoinsRequest{})
+	if err != nil {
+		return fmt.Errorf("get foreign coins: %w", err)
+	}
+
+	for _, coin := range foreignCoinsRes.ForeignCoins {
+		if coin.ForeignChainId != chainID {
+			continue
+		}
+		// sometimes symbol is USDT, sometimes it's like USDT.SEPOLIA
```

Comment:
We will standardize it in the future to always use the later format, I always keep the ERC20 in the name.

I think doing this check is future proof, all existing ERC20 on live network contains the symbol of the underlying

Potential issue is if we have a ERC20 that contains another one in its name, I think this is a potentially rare case

Reply:
Switched to `strings.HasPrefix(coin.Symbol, erc20Symbol) || strings.HasSuffix(coin.Symbol, erc20Symbol)` which should have fewer false positive matches.